SConstruct

```diff
@@ -301,6 +289,12 @@ if selected_platform in platform_list:
     sys.path.insert(0, tmppath)
     import detect
 
+    # Add platform-specific options.
+    for opt in detect.get_opts():
+        opts.Add(opt)
+    opts.Update(env_base)
```

Comment:
We call `opts.Update()` three times in total when a platform is selected, and twice otherwise. I'm not sure if this has any significant impact on build time (doesn't seem to have much for `scons --help` on Linux at least).

On the other hand we generate the `Help()` only once as I remember this as being significant (CC @Xrayez to test and confirm maybe, I remember that all this is slower on Windows).

Reply:
Updating environment and generating help text have little performance impact, it's more about parsing Python scripts, which just happens to be slow on Windows...